ktor-server/ktor-server-plugins/ktor-server-auth/common/src/io/ktor/server/auth/AuthenticationInterceptors.kt

```diff
@@ -69,6 +71,32 @@ public val AuthenticationInterceptors: RouteScopedPlugin<RouteAuthenticationConf
         .findProviders(providers) { it == AuthenticationStrategy.Optional } -
         requiredProviders - firstSuccessfulProviders
 
+    // To cache the request body which can be consumed by the OAuth2 callback handler
+    on(ReceiveBytes) { call, body ->
+        var newBody: Any = body
+
+        val isFormEncoded = call.request.contentType().withoutParameters() == ContentType.Application.FormUrlEncoded
```

Comment:
we can use `match` method instead

Reply:
Can you explain why?

Reply:
for the proper case-insensitive and wildcard handling